test/topology_custom/test_mv_topology_change.py

```diff
@@ -193,3 +194,46 @@ async def test_mv_write_to_dead_node(manager: ManagerClient):
     # will be held for long time until the write timeouts.
     # Otherwise, it is expected to complete in short time.
     await manager.remove_node(servers[0].server_id, servers[-1].server_id, timeout=30)
+
+# Reproduces https://github.com/scylladb/scylladb/issues/21492
+# In this test we perform a write with a view update while the replication factor
+# of the keyspace is being changed and the corresponding base tablet finishes
+# the migration at a different time the view tablet does.
+@pytest.mark.asyncio
+@pytest.mark.parametrize("delayed_replica", ["base", "view"])
+@skip_mode('release', 'error injections are not supported in release mode')
+async def test_mv_rf_change(manager: ManagerClient, delayed_replica: str):
+    servers = await manager.servers_add(2)
+
+    cql = manager.get_cql()
+    await cql.run_async("CREATE KEYSPACE ks WITH replication = {'class': 'NetworkTopologyStrategy', 'replication_factor': 1} AND tablets = {'initial': 1}")
+    await cql.run_async("CREATE TABLE ks.tab (pk int, ck int,  PRIMARY KEY (pk, ck))")
+    await cql.run_async("CREATE MATERIALIZED VIEW ks.tabv AS SELECT pk, ck FROM ks.tab WHERE ck IS NOT NULL PRIMARY KEY (ck, pk)")
+
+    # Insert a row so that there's data to be streamed during tablet migration
+    await cql.run_async("INSERT INTO ks.tab (pk,ck) VALUES (0,0)")
+
+    await asyncio.gather(*[manager.api.enable_injection(s.ip_addr, f"delay_{delayed_replica}_tablet_streaming_by_1s", True) for s in servers])
+
+    res = await cql.run_async("SELECT data_center FROM system.local")
+    this_dc = res[0].data_center
+
+    ks_fut = cql.run_async(f"ALTER KEYSPACE ks WITH replication = {{'class': 'NetworkTopologyStrategy', '{this_dc}':2}}")
+    # Perform a write after the migration has started but before it has finished (there may be just 1 replica due to the injection, so use cl=ONE)
```

Comment:
I will try to play with it later today to reproduce this when decreasing RF, so we can add a case for it too.